snowflake/src/test/java/org/apache/iceberg/snowflake/SnowflakeCatalogTest.java

```diff
@@ -0,0 +1,231 @@
+        "TAB_6",
+        SnowflakeTableMetadata.parseJson(
+            "{\"metadataLocation\":\"gcs://tab6/metadata/v123.metadata.json\",\"status\":\"success\"}"));
+
+    catalog.setSnowflakeClient(client);
+
+    InMemoryFileIO fakeFileIO = new InMemoryFileIO();
+
+    Schema schema =
+        new Schema(
+            Types.NestedField.required(1, "x", Types.StringType.get(), "comment1"),
+            Types.NestedField.required(2, "y", Types.StringType.get(), "comment2"));
+    PartitionSpec partitionSpec =
+        PartitionSpec.builderFor(schema).identity("x").withSpecId(1000).build();
+    fakeFileIO.addFile(
+        "s3://tab1/metadata/v3.metadata.json",
+        TableMetadataParser.toJson(
+                TableMetadata.newTableMetadata(
+                    schema, partitionSpec, "s3://tab1/", ImmutableMap.<String, String>of()))
+            .getBytes());
+    fakeFileIO.addFile(
+        "wasbs://mycontainer@myaccount.blob.core.windows.net/tab3/metadata/v334.metadata.json",
+        TableMetadataParser.toJson(
+                TableMetadata.newTableMetadata(
+                    schema,
+                    partitionSpec,
+                    "wasbs://mycontainer@myaccount.blob.core.windows.net/tab1/",
+                    ImmutableMap.<String, String>of()))
+            .getBytes());
+    fakeFileIO.addFile(
+        "gs://tab5/metadata/v793.metadata.json",
+        TableMetadataParser.toJson(
+                TableMetadata.newTableMetadata(
+                    schema, partitionSpec, "gs://tab5/", ImmutableMap.<String, String>of()))
+            .getBytes());
+
+    catalog.setFileIO(fakeFileIO);
+
+    Map<String, String> properties = Maps.newHashMap();
+    catalog.initialize(TEST_CATALOG_NAME, properties);
+  }
+
+  @Test
+  public void testListNamespace() {
+    List<Namespace> namespaces = catalog.listNamespaces();
+    Assert.assertEquals(
+        Lists.newArrayList(
+            Namespace.of("DB_1", "SCHEMA_1"),
+            Namespace.of("DB_2", "SCHEMA_2"),
+            Namespace.of("DB_3", "SCHEMA_3"),
+            Namespace.of("DB_3", "SCHEMA_4")),
+        namespaces);
+  }
+
+  @Test
+  public void testListNamespaceWithinDB() {
+    String dbName = "DB_1";
+    List<Namespace> namespaces = catalog.listNamespaces(Namespace.of(dbName));
+    Assert.assertEquals(Lists.newArrayList(Namespace.of(dbName, "SCHEMA_1")), namespaces);
+  }
+
+  @Test
+  public void testListNamespaceWithinNonExistentDB() {
+    // Existence check for nonexistent parent namespaces is optional in the SupportsNamespaces
+    // interface.
+    String dbName = "NONEXISTENT_DB";
+    Assert.assertThrows(RuntimeException.class, () -> catalog.listNamespaces(Namespace.of(dbName)));
+  }
+
+  @Test
+  public void testListNamespaceWithinSchema() {
+    // No "sub-namespaces" beyond database.schema; invalid to try to list namespaces given
+    // a database.schema.
+    String dbName = "DB_3";
+    String schemaName = "SCHEMA_4";
+    Assert.assertThrows(
```

Comment:
same as above. Also please add a `.hasMessage(..)` check to make sure we're getting the right message here.

Reply:
Done. Switched to `Assertions.assertThatExceptionOfType(...).withMessageContaining(...)` here and elsewhere.